napari_arboretum/tree.py

```diff
@@ -30,28 +32,27 @@ class Edge:
     id: Optional[int] = None
 
 
-def _build_tree(nodes) -> Tuple[List[Edge], List[Annotation]]:
+def layout_tree(nodes: List[TreeNode]) -> Tuple[List[Edge], List[Annotation]]:
     """Build and layout the edges of a lineage tree, given the graph nodes.
 
     Parameters
     ----------
-    nodes : list
+    nodes :
         A list of graph.TreeNode objects encoding a single lineage tree.
 
     Returns
     -------
-    edges : list
+    edges :
         A list of edges to be drawn.
-
-    annotations : list
+    annotations :
         A list of annotations to be added to the graph.
     """
     # put the start vertex into the queue, and the marked list
     root = nodes[0]
 
     queue = [root]
     marked = [root]
-    y_pos = [0]
+    y_pos = [0.0]
```

Comment:
Seeing a "position" like this looks a bit odd as I'd have naively assumed that it was an index. What are the advantages of forcing it be a float here?

Reply:
This is to satisfy `mypy`. I'm guessing that somewhere else a float is appended to `y_pos`, so `mypy` wants to make sure the list is not a mix of floats and integers. This isn't an index, it's the location along an axis, so I think makes sense that it's floating point?

Reply:
Sounds good